haystack/nodes/answer_generator/openai.py

```diff
@@ -94,6 +94,7 @@ def __init__(
         else:
             self.MAX_TOKENS_LIMIT = 2048
 
+    @retry_with_exponential_backoff(initial_delay=10, max_retries=3)
```

Comment:
Should we allow custom values of `max_retries` here?

Reply:
But they are custom, these are parameters @bogdankostic

Reply:
But isn't this hard-coded to three in any case, so not customizable by the user?

Reply:
Aha, I see what you mean now. No, unfortunately, we can't use some self attributes or other variables here. They have to be explicitly set. But who would want to do that, anyway? Ok, OpenAI retry back off might change - in that case, we can change it.

Reply:
Yeah, I think we could make `max_retries` an initialization parameter to let the user specify the number of retries they would be willing to make. So then the decorator would look like 
```python
@retry_with_exponential_backoff(initial_delay=10, max_retries=self.max_retries)
```

Reply:
I wanted to do that @sjrl, but python doesn't support self references in decorators

Reply:
Well, I learned that today myself :-)

Reply:
I also thought `self.max_retries` would work there, I guess all three of us learned this today :D